geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegionQueue.java

```diff
@@ -71,10 +73,10 @@
   private final Map indexes;
 
   /**
-   * A transient queue to maintain the eventSeqNum of the events that are to be sent to remote site.
-   * It is cleared when the queue is cleared.
+   * A transient deque, but should be treated like as a fifo queue to maintain the eventSeqNum of
+   * the events that are to be sent to remote site. It is cleared when the queue is cleared.
    */
-  private final BlockingQueue<Object> eventSeqNumQueue = new LinkedBlockingQueue<Object>();
+  private final BlockingDeque<Object> eventSeqNumDeque = new LinkedBlockingDeque<Object>();
```

Comment:
Does this really have to be a Deque of type Object?? Don't we know what type of objects we are putting in here?